### PR TITLE
Test Framework updates

### PR DESCRIPTION
**Pull request by rstens**
# Description

Added framework updates to work with Faker for realistic but fake test data.
Add other randomizers for fields that faker does not support.
Updated scripts and work on better element recognition.
Add the log in capability for our roles.
Create set/get/do functions for common fields

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-vmvfjv-tools.pathfinder.gov.bc.ca/dashboard?id=common-forms-toolkit-pr-140

File: tests/functional-tests/src/test/groovy/pages/app/MinesOperatorPage.groovy [hunk 1]
```diff
@@ -1,119 +1,170 @@
...
+      //Housekeeping for Ill Workers
+      infectedHousekeeping { $("input", type:"checkbox", "data-test":"cb-form-infectedHousekeeping",dataState).parentsUntil("div", class:"v-input__slot") }
+
+      //Waste Management for Ill Workers
+      infectedWaste { $("input", type:"checkbox", "data-test":"cb-form-infectedWaste",dataState).parentsUntil("div", class:"v-input__slot") }
+
+      certifyAccurateInformation { $("input", type:"checkbox", "data-test":"cb-form-certifyAccurateInformation",dataState).parentsUntil("div", class:"v-input__slot") }
+      agreeToInspection { $("input", type:"checkbox", "data-test":"cb-form-agreeToInspection",dataState).parentsUntil("div", class:"v-input__slot") }
+
+      subMit { $("button", class: "v-btn v-btn--contained theme--light v-size--default primary", "data-test":"btn-form-submit") }
```
Comment (jujaga) [inline tests/functional-tests/src/test/groovy/pages/app/MinesOperatorPage.groovy:166]: Guessing you didn't mean to capitalize the m in submit

File: tests/functional-tests/src/test/groovy/pages/app/base/BaseAppPage.groovy [hunk 2]
```diff
@@ -12,15 +13,7 @@ import modules.FooterModule
  */
 class BaseAppPage extends Page {
   static content = {
-    headerModule { module(HeaderModule) }
+     headerModule { module(HeaderModule) }
```
Comment (jujaga) [inline tests/functional-tests/src/test/groovy/pages/app/base/BaseAppPage.groovy:16]: Extra whitespace indentation

File: tests/functional-tests/src/main/groovy/listeners/BrowserStackReportingSpec.groovy [hunk 3]
```diff
@@ -20,11 +20,11 @@ class BrowserStackReportingSpec extends GebReportingSpec {
     _browser
   }
 
-  @Override
+   @Override
```
Comment (jujaga) [inline tests/functional-tests/src/main/groovy/listeners/BrowserStackReportingSpec.groovy:23]: Extra whitespace indentation

File: tests/functional-tests/src/test/groovy/specs/ComfortHomeSpec.bkp [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+As a user
+I want use the comfort application
+So that I can manage my forms
+""")
+@Issue("https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-1292")
+@Stepwise
+class ComfortHomeSpec extends BaseSpec implements Login {
+
+  @Unroll
+/*  def "As a None user, I wan to to log in into the Comfort application."() {
```
Comment (jujaga) [inline tests/functional-tests/src/test/groovy/specs/ComfortHomeSpec.bkp:21]: Want missing t

File: tests/functional-tests/src/test/groovy/specs/MinesOperatorSpec.groovy [hunk 5]
```diff
@@ -1,80 +1,82 @@
...
+          covidEmail.value(randomEmail())
 
           //Provide your accommodation details
           startDate.click()
-          //Thread.sleep(500)
-          startDate.jquery.removeAttr("readonly") //Nasty Hack to make the dates working
-          startDate << "2020-06-15"
+          basestartDate.jquery.removeAttr("readonly") //Nasty Hack to make the dates working
+          basestartDate << "2020-07-15"
+          basestartDate.jquery.attr("readonly","readonly") //Nasty Hack to make the dates working
```
Comment (jujaga) [inline tests/functional-tests/src/test/groovy/specs/MinesOperatorSpec.groovy:50]: Does the infrastructure have the ability to run the more JS standard .addAttribute() and .removeAttribute() functions or does it need to leverage jQuery to work?

File: tests/functional-tests/src/test/groovy/pages/app/MinesOperatorPage.groovy (hunk 1 above)
Comment (rstens) [reply tests/functional-tests/src/test/groovy/pages/app/MinesOperatorPage.groovy]: Needs a capital, but created another identifier

File: tests/functional-tests/src/test/groovy/specs/MinesOperatorSpec.groovy (hunk 5 above)
Comment (rstens) [reply tests/functional-tests/src/test/groovy/specs/MinesOperatorSpec.groovy:50]: Standard JS does not work in the test framework for this purpose. The jQuery function does not get all confused by Vue.
I'd like to have the JS possibility, but after I spent a full day on trying to make that work, I resigned to using jQuery.

File: tests/functional-tests/src/test/groovy/specs/ComfortHomeSpec.bkp (hunk 4 above)
Comment (rstens) [reply tests/functional-tests/src/test/groovy/specs/ComfortHomeSpec.bkp]: Done

**Pull request merged**